app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -47,6 +52,13 @@ class MetaRPCClient {
   close() {
     this.notificationChannel.removeAllListeners();
     this.uncaughtErrorChannel.removeAllListeners();
+    // fail all unfinished requests
+    for (const [id, handler] of this.requests) {
```

Comment:
we should PR this seperately - nice fixup

Reply:
We can't have the queue logic without it. I can split it out, and we merge it first.

Reply:
Check out https://github.com/MetaMask/metamask-extension/pull/15570